packages/mds-policy-service/@types/intents/index.ts

```diff
@@ -2,61 +2,120 @@ import type { BaseRule } from '../models'
 import type { BaseIntentPolicyUserFields, INTENT_TYPE, PartialMicroMobilityStatesMap } from './common'
 import { COMMON_INTENT_RULE_DEFAULTS } from './common'
 
-export type IntentRuleDefaults = Omit<BaseRule<PartialMicroMobilityStatesMap>, 'rule_id'>
+export type IntentRuleDefaults = Omit<BaseRule<PartialMicroMobilityStatesMap>, 'rule_id' | 'modality'> & {
+  modality: 'micromobility'
+}
 
 export const NO_PARKING_INTENT_RULE_CONSTANTS: IntentRuleDefaults = {
+  ...COMMON_INTENT_RULE_DEFAULTS,
+  modality: 'micromobility',
   name: 'No Parking Zone',
   maximum: 0,
-  states: { available: [], non_operational: [], reserved: [], unknown: [] },
-  ...COMMON_INTENT_RULE_DEFAULTS
+  states: { available: [], non_operational: [], reserved: [], unknown: [] }
 }
+
 export const PERMITTED_VEHICLE_COUNT_INTENT_RULE_CONSTANTS: IntentRuleDefaults = {
+  ...COMMON_INTENT_RULE_DEFAULTS,
+  modality: 'micromobility',
   maximum: 0,
   minimum: 0,
   name: 'Permitted Vehicle Count',
-  states: {},
-  ...COMMON_INTENT_RULE_DEFAULTS
+  states: {}
+}
+
+export const PARKING_TIME_LIMIT_INTENT_RULE_CONSTANTS: IntentRuleDefaults = {
+  ...COMMON_INTENT_RULE_DEFAULTS,
+  modality: 'micromobility',
+  rule_units: 'minutes',
+  rule_type: 'time',
+  maximum: 0,
+  minimum: 0,
+  name: 'Parking Time Limit',
+  states: { available: [], non_operational: [], reserved: [] }
+}
+
+export const REQUIRED_PARKING_INTENT_RULE_CONSTANTS: IntentRuleDefaults = {
+  ...COMMON_INTENT_RULE_DEFAULTS,
+  modality: 'micromobility',
+  maximum: 0,
+  minimum: 0,
+  name: 'Required Parking',
+  rule_units: 'minutes',
+  states: { available: [], non_operational: [], reserved: [] }
 }
 
 export const INTENT_RULE_CONSTANTS: { [K in INTENT_TYPE]: IntentRuleDefaults } = {
   no_parking: NO_PARKING_INTENT_RULE_CONSTANTS,
-  permitted_vehicle_count: PERMITTED_VEHICLE_COUNT_INTENT_RULE_CONSTANTS
+  permitted_vehicle_count: PERMITTED_VEHICLE_COUNT_INTENT_RULE_CONSTANTS,
+  parking_time_limit: PARKING_TIME_LIMIT_INTENT_RULE_CONSTANTS,
+  required_parking: REQUIRED_PARKING_INTENT_RULE_CONSTANTS
 }
 
-export type IntentDraft<
-  I extends INTENT_TYPE,
-  INTENT_POLICY_FIELDS extends BaseIntentPolicyUserFields = NoParkingIntentPolicyUserFields,
-  INTENT_RULE_FIELDS extends BaseIntentRuleUserFields = NoParkingIntentRuleUserFields
-> = {
+export type IntentDraft<I extends INTENT_TYPE> = {
   intent_type: I
-  policy_fields: INTENT_POLICY_FIELDS
-  rule_fields: INTENT_RULE_FIELDS
+  policy_fields: BaseIntentPolicyUserFields
+  rule_fields: I extends 'no_parking'
+    ? NoParkingIntentRuleUserFields
+    : I extends 'permitted_vehicle_count'
+    ? PermittedVehicleCountIntentRuleUserFields
+    : I extends 'parking_time_limit'
+    ? ParkingTimeLimitIntentRuleUserFields
+    : I extends 'required_parking'
+    ? RequiredParkingIntentRuleUserFields
+    : NoParkingIntentRuleUserFields
```

Comment:
Should the last fallback type be `never` here to prevent users from supplying a NoParkingIntent rule to a different intent type? Should only be possible if we add an intent type and then forget to extend this type but still useful for defensive programming.

Reply:
Ah thank you! I wasn't sure what to put down, but that makes the most sense!